### ISSUE TITLE
MQTT Comunnication

### ISSUE THREAD

**Issue by ladislav37** [sidoh/esp8266_milight_hub#280]
Hello.
Today I tray to connect this hub with OpanHab trought MQTT and:
Setup according your docs.. (Devide ID: 0x6047, type: cct, group: 4)
![image](https://user-images.githubusercontent.com/39069162/39727337-97571428-5252-11e8-8d2a-c3ea1b353fa9.png)

I use mqtt-spy to publish on mqtt broaker (Mosquitto@local Ip..)
![image](https://user-images.githubusercontent.com/39069162/39727437-f369268e-5252-11e8-8bb6-42ff1738c4d8.png)

And answer (direct _sub from rpi.., I try to send only {"state":"OFF"} but the result was same..)
![image](https://user-images.githubusercontent.com/39069162/39727480-25d74222-5253-11e8-94c3-93488fe5b254.png)

But this was send from hub on RF..
![image](https://user-images.githubusercontent.com/39069162/39727519-4b743e40-5253-11e8-9269-dae12223d0ec.png)

As you can see, no ID, no group, different controller type. Could you please help me with this. I making some mistake or it is FW bug? Is it possible to set device ID/group/type in MQTT data, to be possible send command into any receiver from one hub?

Thank you for any answer/help.

**Comment by sidoh** [comment]
MQTT configs are a bit convoluted.  There's detail in the README:

https://github.com/sidoh/esp8266_milight_hub#more-detail-on-mqtt_topic_pattern

tl;dr, update your configs to:

```json
{
  "mqtt_topic_pattern": "milight/commands/:device_id/:device_type/:group_id",
  "mqtt_update_topic_pattern": "milight/updates/:device_id/:device_type/:group_id",
  "mqtt_state_topic_pattern": "milight/states/:device_id/:device_type/:group_id"
}
```

Going to close, but please feel free to reopen if I missed something.
